moped-database/migrations/1700149643858_parent_audit_values/up.sql

```diff
@@ -0,0 +1,73 @@
+-- Altering moped_project table to add a new column for tracking the user who last updated a record
+ALTER TABLE moped_project ADD COLUMN updated_by_user_id INTEGER DEFAULT NULL;
+COMMENT ON COLUMN moped_project.updated_by_user_id IS 'User ID of the person who last updated the project';
+
+-- Creating or replacing a function to update audit logs in parent records
+CREATE OR REPLACE FUNCTION update_parent_records_audit_logs()
+RETURNS TRIGGER AS $$
+DECLARE
+  project_id_variable INTEGER;
+  user_id_variable INTEGER;
+  query TEXT;
+BEGIN
+  
+  -- Constructing a dynamic query to select the user ID from the triggered table
+  query := 'SELECT updated_by_user_id FROM ' || quote_ident(TG_TABLE_NAME) || 
+           ' WHERE id' || ' = ' || NEW.id || ';';
+  
+  -- Executing the dynamic query
+  EXECUTE query INTO user_id_variable;
```

Comment:
this is a TIL for me for sure! in this case, instead of querying this value from the trigger table, i think it can be retrieved from `NEW.updated_by_user_id`?

Reply:
Oh yes, of course! Thank you - that is so the correct way to do this instead of constructing a dynamic query. I'm used to just having an ID to start with and forgot we had the rest of the row as well. 💯  This is addressed in commit baba803d224b27dcb812d75ebf9144c8d589cabc.